src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -20,18 +27,37 @@ public void Initialize(TelemetryConfiguration configuration)
 
             _listener = new ActivityListener
             {
-                ShouldListenTo = source => source.Name.StartsWith("Microsoft.Azure.Functions.Worker"),
+                ShouldListenTo = source => source.Name.StartsWith(TraceConstants.FunctionsActivitySource),
                 ActivityStarted = activity =>
                 {
                     var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(activity);
```

Comment:
I think it would be helpful to inspect the `ActivityKind` and instantiate an appropriate telemetry item from there.

Reply:
I'd vote that we do that when (if) we add new activities. I'm really hoping that this assembly is a short-lived "bridge" solution until the Azure Monitor otel support GAs -- then we can recommend folks go there instead.

Reply:
Going to re-open this for reconsideration given our discussion of using `ActivityKind=Server` instead of `Internal`. In which case, should we use `RequestTelemetry` for `Server`?

Regarding the app map UX being messed up if we use request telemetry, as the host will already be emitting that. Should host and worker be seen as two separate apps? This way the UX will draw lines between the Host and Worker.

Reply:
So this part (the App Insights package) will still be in Preview -- I'd like to get this in and we can come back to it after getting some feedback.